sunscreen_compiler_common/src/context.rs

```diff
@@ -258,6 +268,17 @@ where
      * Add a node to the parse graph.
      */
     pub fn add_node(&mut self, operation: O) -> NodeIndex {
+        #[cfg(feature = "debugger")]
+        {
+            let group_id = self.group_counter;
+            self.group_counter += 1;
```

Comment:
Probably don't increase the group counter each time you make a node, but rather when you push a group onto the stack that doesn't already exist in your group data structure.